src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/ProxyHeaderModule.cs

```diff
@@ -42,10 +46,15 @@ public void Init(HttpApplication context)
 
         if (options.UseForwardedHeaders)
         {
+            _wrField = typeof(HttpRequest).GetField("_wr", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
I'd prefer some sort of helper class/struct that would handle all the reflection stuff behind the scenes so the module doesn't have to feel so "hacky"

Reply:
Yeah makes sense now that we call this in two places, done

Reply:
Oh, are you talking about initializing the fields/type I guess?

Reply:
I mean I can move this into some kind of InitializeReflection if you really want

Reply:
So if I hear you, you prefer some kind of new HostReflectionHelper class here that we can pass in the request

Reply:
I'd prefer the helper class direction